### PR TITLE
Python: Module not intended for production

### PR DESCRIPTION
**Pull request by dilanbhalla**
Query to detect use of the http.server module, which has been given a security warning for production use within the Python 3 official documentation, as it only implements basic security checks.

### CODE REVIEW

File: python/ql/src/Security/CWE-602/DangerousHttpServer.ql [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    and
+    msg = "http.server is not recommended for production. It only implements basic security checks."
+    and
+    result = Module::named(mod)
+}
+
+from AstNode c, string mod, string msg
+where
+    c = http_server(mod, msg).getAReference().getNode()
+select c, msg
```
Comment (intrigus-lgtm) [inline python/ql/src/Security/CWE-602/DangerousHttpServer.ql:26]: Why not simplify this?
```suggestion
from AstNode c
where
    c = Module::named("http.server").getAReference().getNode()
select c, "http.server is not recommended for production. It only implements basic security checks."
```

Comment (dilanbhalla) [reply python/ql/src/Security/CWE-602/DangerousHttpServer.ql:26]: You're right! I was originally going to add more to this query which would have required the structure I used originally. But this is much more concise.

**Comment by intrigus-lgtm** [comment]
I think you should move your query to the `experimental` folder as stated [here](https://github.com/github/codeql/blob/master/CONTRIBUTING.md)

Also see this existing comment:
https://github.com/github/codeql/pull/3396#pullrequestreview-415419534

**Comment by RasmusWL** [comment]
I looked into this PR now. It's true that using `python -m http.server` or using unmodified `http.server.SimpleHTTPRequestHandler` as a handler pose some security problems, and should not be used in production.

However, it does seem a bit too broad to alert on _any_ use of the `http.server` module. My understanding is that using your own `BaseHTTPRequestHandler` subclass to handle requests does not have the same security problems.

So I'm going to close this PR, since it's simply too broad. Sorry for the long delay in responding.

**Pull request closed without merging**